crates/pathfinder/src/rpc/v02/method/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,176 @@
+use anyhow::Context;
+
+use crate::core::StarknetTransactionHash;
+use crate::rpc::v02::types::reply::Transaction;
+use crate::rpc::v02::RpcContext;
+use crate::storage::StarknetTransactionsTable;
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct GetTransactionByHashInput {
+    transaction_hash: StarknetTransactionHash,
+}
+
+crate::rpc::error::rpc_error_subset!(GetTransactionByHashError: TxnHashNotFound);
+
+#[allow(dead_code)]
+pub async fn get_transaction_by_hash(
+    context: std::sync::Arc<RpcContext>,
+    input: GetTransactionByHashInput,
+) -> Result<Transaction, GetTransactionByHashError> {
+    if let Some(pending) = &context.pending_data {
+        let pending_tx = pending.block().await.and_then(|block| {
+            block
+                .transactions
+                .iter()
+                .find(|tx| tx.hash() == input.transaction_hash)
+                .cloned()
+        });
+
+        if let Some(pending_tx) = pending_tx {
+            return Ok(pending_tx.into());
+        }
+    }
+
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    let jh = tokio::task::spawn_blocking(move || {
+        let _g = span.enter();
+        let mut db = storage
+            .connection()
+            .context("Opening database connection")?;
+
+        let db_tx = db.transaction().context("Creating database transaction")?;
+
+        // Get the transaction from storage.
+        StarknetTransactionsTable::get_transaction(&db_tx, input.transaction_hash)
```

Comment:
I imagine this call is used in multiple places?

(Thinking about inlining the sql calls within methods directly).

Not something we have to focus on just yet, if we do at all.

Reply:
I think for now a _lot_ of these functions are used at least from the old and the new RPC methods...